### ISSUE TITLE
Texconv fails to read DXT compressed file

### ISSUE THREAD

**Issue by JT8D-17** [Microsoft/DirectXTex#79]
When trying to read a DXT1/3/5 compressed texture file, Texconv shows the following error:

_[filename] FAILED (88982f07)_

The same error message also appears when tryng to read the file with Texdiag.

Converting the file to 32bit and then trying again works fine.

The file in question is attached below.

[sample.zip](https://github.com/Microsoft/DirectXTex/files/1733219/sample.zip)

**Comment by walbourn** [comment]
That BMP is not supported by Windows Imaging Component (WIC). If you try viewing it in the Windows Photo app, it also fails there.


What created that BMP file?


Trying my BMPDUMP tool I see:

BMP c:\Users\Chuck\Desktop\sample.bmp
        INFO header (40 bytes)
1024 x 1024
1 planes
16 bits
861165636 compression type
1048576 image size
0 x 0 pels per meter
0 colors used
0 colors important



-Chuck Walbourn
Senior SDE
http://blogs.msdn.com/chuckw
________________________________

**Comment by walbourn** [comment]
That `BMP` is not supported by Windows Imaging Component (WIC). If you try viewing it in the Windows Photo app, it also fails there.

What created that ``BMP`` file?

Trying my [BMPDUMP](https://raw.githubusercontent.com/wiki/Microsoft/DirectXTex/bmpdump.cpp) tool I see:

    BMP c:\Users\Chuck\Desktop\sample.bmp
            INFO header (40 bytes)
    1024 x 1024
    1 planes
    16 bits
    861165636 compression type
    1048576 image size
    0 x 0 pels per meter
    0 colors used
    0 colors important

The compression type is not one of:

    BI_RGB (0)
    BI_RLE8 (1)
    BI_RLE4 (2)
    BI_BITFIELDS (3)
    BI_JPEG (4)
    BI_PNG (5)

**Comment by JT8D-17** [comment]
Thanks for the reply.
Yes, both image viewer apps delivered with Windows 10 fail to display it. This is normal.
Nconvert, however, has no problems reading the file.

I'm not the original author of the file, so I can only make an assumption which tool was used to compress it. Since it's a texture for use in Microsoft Flight Simulator/Lockheed Martin Prepar3D, there's only two possibilites. Either ACES' ImageTool, shipped with the Flight Simulator SDK or DXTBMP.
The latter is available here: http://www.mwgfx.co.uk/programs/dxtbmp.htm
The former can't be obtained without installing the rest of the MSFS/P3D SDK, so I've attached it here for convienience. (May be deleted after the next reply by you or me.)
[ImageTool.zip](https://github.com/Microsoft/DirectXTex/files/1733773/ImageTool.zip)

Either tool fails to produce a Texconv copliant file when saving with BC1/DXT1 to BC3/DXT5 compression. Only files saved in 32bit by either tool is correctly read by Texconv.
On the other hand, a BC1/2/5 compressed file produced by Texconv can not be read by ImageTool. DXTBMP has no problems either way.

When in reading a 32bit file produced by ImageTool or DXTBMP, reading the file with Texconv shows that it's read as B8G8R8X8_UNORM. Converting this file into R8B8G8A8_UNORM format (DDS filetype) again produces a file that can not be read by Imagetool. Only selecting B8G8R8A8_UNORM as output format produces a usable file. So I assume that ImageTool/DXTBMP saves files with a non-regular order of channels and that this is what throws WIC and Texconv off.

Hope this helps.

**Comment by walbourn** [comment]
OK, I now see what's happening here. Someone in the FlightSim team got clever and decided to containerize a DXTn texture in a ``BMP`` rather than used the long-established ``DDS`` format. It's not an officially supported way to write a DXTn, but I might be able to do something about that...

**Comment by walbourn** [comment]
I added support for loading these Extended ``BMP`` files, but I don't support writing them.

See [this commit](https://github.com/Microsoft/DirectXTex/commit/06816df22c2c365930963021bd3910c8d1920d15).

**Comment by JT8D-17** [comment]
I would support writing to such a weird format either, to be honest.
Reading these files is all I need and I'll check that feature out in the next release.

I've also edited my previous post to remove the attachment and iron out some mistakes.
For example, I don't know why I stated that BCn compression produces files that are not Imagetool compliant. I've just checked the results from a batch texture conversion to BC2_UNORM (DXT3) DDS again and randomly picked samples opened in Imagetool just fine.

You're correct on the vertical flip requirement and yes, -vflip does the job.

Thank you for investigating and implementing this!
